File: python/questao5.py
def print_reversed_string_with_a_replaced(string):
    reversed_string = string[::-1]
    replaced_string = ''.join('*' if char == 'A' else char for char in reversed_string)
    print(replaced_string)

def main():
    phrase1 = input("Digite a primeira frase: ")
    phrase2 = input("Digite a segunda frase: ")

    print("Frase 1 invertida e com 'A' trocado por '*': ", end='')
    print_reversed_string_with_a_replaced(phrase1)

    print("Frase 2 invertida e com 'A' trocado por '*': ", end='')
    print_reversed_string_with_a_replaced(phrase2)

if __name__ == "__main__":
    main()

File: c/questao5.c
#include <stdio.h>
#include <string.h>

void printReversedStringWithAReplaced(char* str) {
    int length = strlen(str);

    for (int i = length - 1; i >= 0; i--) {
        if (str[i] == 'A') {
            printf("*");
        } else {
            printf("%c", str[i]);
        }
    }
    printf("\n");
}

int main() {
    char phrase1[100], phrase2[100];

    printf("Digite a primeira frase: ");
    fgets(phrase1, sizeof(phrase1), stdin);

    printf("Digite a segunda frase: ");
    fgets(phrase2, sizeof(phrase2), stdin);

    printf("Frase 1 invertida e com 'A' trocado por '*': ");
    printReversedStringWithAReplaced(phrase1);

    printf("Frase 2 invertida e com 'A' trocado por '*': ");
    printReversedStringWithAReplaced(phrase2);

    return 0;
}
